File: scripts/commit_integration.py
#!/usr/bin/env python3
import subprocess
import sys
import os

# Change to project directory
os.chdir('/vercel/share/v0-project')

try:
    # Stage all changes
    print("📝 Staging changes...")
    subprocess.run(['git', 'add', '-A'], check=True)
    
    # Create commit
    print("✅ Creating commit...")
    commit_message = """feat: Integrate coach dashboard with backend API

- Added dynamic data loading for entrepreneurs and sessions
- Enhanced coach.service.ts with new API methods:
  * getDashboardStats() - Load dashboard statistics
  * getCoachEntrepreneurs() - Load assigned entrepreneurs
  * getUpcomingSessions() - Load upcoming sessions
  * getDashboardOverview() - Combined data endpoint
- Updated CoachDashboard component to use backend data
- Replaced static mockup data with dynamic content
- Added loading states and empty states
- Added DTOs for type-safe data handling
- Updated styles for session statuses (green, orange, red)
- Added documentation and integration guide

Co-authored-by: v0[bot] <v0[bot]@users.noreply.github.com>"""
    
    subprocess.run(
        ['git', 'commit', '-m', commit_message],
        check=True
    )
    
    print("✨ Commit successful!")
    print("📤 Pushing to frontend-integration branch...")
    subprocess.run(['git', 'push', 'origin', 'frontend-integration'], check=True)
    print("✅ Push successful!")
    
except subprocess.CalledProcessError as e:
    print(f"❌ Error: {e}")
    sys.exit(1)

File: scripts/commit-integration.js
import { execSync } from 'child_process';
import { existsSync, mkdirSync } from 'fs';
import { resolve } from 'path';

const projectDir = '/vercel/share/v0-project';

try {
  console.log('📁 Navigating to project directory...');
  process.chdir(projectDir);

  console.log('📦 Staging all changes...');
  execSync('git add -A', { stdio: 'inherit' });

  console.log('✍️  Creating commit for coach dashboard integration...');
  const commitMessage = `feat: integrate coach dashboard with backend APIs

- Add dynamic data loading for entrepreneurs list
- Add dynamic data loading for upcoming sessions
- Add dashboard stats endpoint integration
- Create new DTOs for CoachEntrepreneur, DashboardStats, UpcomingSession
- Implement loadCoachEntrepreneurs() and loadUpcomingSessions() methods
- Update CoachDashboard component to use CoachService
- Replace hardcoded data with dynamic backend data
- Add loading states and empty state handling
- Add responsive UI for dynamic data display

The coach dashboard now dynamically loads all data from the backend instead of using static mockup data. This includes:
- Coach's assigned entrepreneurs with completion rates
- Upcoming sessions with status and meeting links
- Dashboard statistics (meetings, tasks, phases, projects)

All data is fetched on component initialization and displays proper loading/error states.`;

  execSync(`git commit -m "${commitMessage.replace(/"/g, '\\"')}"`, { stdio: 'inherit' });

  console.log('✅ Commit completed successfully!');
  console.log('\n📝 Changes committed to the frontend-integration branch.');
} catch (error) {
  console.error('❌ Error during commit:', error.message);
  process.exit(1);
}
